### PR TITLE
CU-3qveuku - doc - XCVM use cases - cross chain swap

### PR DESCRIPTION
**Pull request by hussein-aitlahcen**
- introduce doc content for xcvm use cases
- introduce cross chain swap example

### CODE REVIEW

File: docs/docs/products/xcvm/use-cases/swap.mdx [hunk 1]
```diff
@@ -0,0 +1,112 @@
+# Cross-Chain Swap
+
+:::tip
+
+Developers should be interested to read [the XCVM
+specification.](https://github.com/ComposableFi/composable/blob/main/code/xcvm/SPEC.md)
```
Comment (KaiserKarel) [inline docs/docs/products/xcvm/use-cases/swap.mdx:6]: Let's make this a permalink so it doesn't break

File: docs/docs/products/xcvm/use-cases/swap.mdx [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+Decentralized exchanges are one of (if not the one) the most widely used DeFi
+protocol in the blockchain ecosystem. It allows (through a contract) anyone to
+atomically exchange a token **A** against a token **B** at a defined rate (such
+as with the invariant `x * y = k`). This operation is executed atomically from
+within a transaction and we call this action a **swap**.
+
+:::
+
+In this example, we will execute a cross-chain swap on top of XCVM and
```
Comment (KaiserKarel) [inline docs/docs/products/xcvm/use-cases/swap.mdx:34]:
```suggestion
In this example, we will execute a cross-chain swap on through XCVM and
```

**Review by KaiserKarel — APPROVED** [review]
Great guide

**Review by dzmitry-lahoda — APPROVED** [review]

**Review by cor — APPROVED** [review]
LGTM

**Pull request merged**